### PR TITLE
Use specific commit SHAs to refer to builder images

### PR DESCRIPTION
**Pull request by dgn**
Currently, we set `imagePullPolicy: Always` to make sure we're always using the latest builder images. This causes an up to 30s delay for each job, because prow has to pull the image first. The better solution is to use unique tags like `2.0-${COMMIT_SHA}` so that prow can reuse an image if it has been pulled before.

There's one disadvantage to this solution, and that is the fact that we're now pushing all containers at once if one of them changes. We could implement it the same way as before, but I think it's helpful to have a full set of images per commit SHA - this way, you don't have to remember which commit introduced proxy-builder changes and which introduced maistra-builder changes - if there's one image with that SHA, all the other images are present, too.

Note that I've also removed the 1.2 builder image

### CODE REVIEW

File: tools/bump_builder.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
```
Comment (jwendell) [inline tools/bump_builder.sh:1]: I think we could simplify this logic and erhaps even remove this file if we had the SHA in a single location, defined in a variable and being used everywhere.

as a reference, this is upstream: https://github.com/istio/common-files/blob/master/files/common/scripts/setup_env.sh#L62

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,5 @@
 HUB ?= quay.io/maistra-dev
+SHA ?= $(shell git rev-parse --short=8 HEAD)
```
Comment (jwendell) [inline Makefile:2]: istio uses date + branch: https://github.com/istio/tools/blob/master/docker/build-tools/build-and-push.sh#L29
can we use the same?

**Review by jwendell — COMMENTED** [review]
I think the gain is minimum, right? I mean, kubelet will not download the image every time, it will check if the image resolves to a SHA not previously downloaded. I'm ok with the changes if they save a few seconds, we should just make sure to not complicate things too much :)

I'm also thinking on unifying the images (not having a separate image for the proxy). I'm playing with it and can submit a pr later on.

**Comment by dgn** [comment]
> I think the gain is minimum, right? I mean, kubelet will not download the image every time, it will check if the image resolves to a SHA not previously downloaded. I'm ok with the changes if they save a few seconds, we should just make sure to not complicate things too much :)

You're probably right on this. I guess I'll do some more benchmarking to see where the bottlenecks actually are in terms of job startup times.

> I'm also thinking on unifying the images (not having a separate image for the proxy). I'm playing with it and can submit a pr later on.

Cool, I think that makes sense - if it's easy to do. With envoy being so picky about the build env, I'm not sure it'll work in the long run.

File: Makefile (hunk 2 above)
Comment (dgn) [reply Makefile:2]: we can add date, but I'm not sure about branch - we keep everything on the `main` right now

**Pull request closed without merging**